README

iOS Ports Example
=================

   iOS Ports Example is a simple iOS application which demostrates reference
   iOS Ports, link to a ported library, how to use iOS Ports Library to
   provide acknowledgements for software used in an application.


Disclaimer
==========

   This software is provided by the copyright holders and contributors "as
   is" and any express or implied warranties, including, but not limited to,
   the implied warranties of merchantability and fitness for a particular
   purpose are disclaimed. In no event shall Bindle Binaries be liable for
   any direct, indirect, incidental, special, exemplary, or consequential
   damages (including, but not limited to, procurement of substitute goods or
   services; loss of use, data, or profits; or business interruption) however
   caused and on any theory of liability, whether in contract, strict
   liability, or tort (including negligence or otherwise) arising in any way
   out of the use of this software, even if advised of the possibility of
   such damage.


Software Requirements
=====================

   Git 1.7.2.3
   Mac OS X 10.6.7
   Xcode 4.0
   iOS SDK 4.3
   iOS Ports Library 0.3


Maintainers
===========

   David M. Syzdek
   Bindle Binaries
   syzdek@bindlebinaries.com


Community
=========

   Mailing List:
      ios-ports@lists.bindlebinaries.com
      https://lists.bindlebinaries.com/mailman/listinfo/ios-ports


Source Code
===========

The source code for this project is maintained using git (http://git-scm.com).
The following contains information to checkout the source code from the git
repository.

   Git URLs:
      git://github.com/bindle/iOSPortsExample.git
      git://git.scm.bindlebinaries.com/pub/scm/com.bindlebinaries/iOSPortsExample.git
      http://http.scm.bindlebinaries.com/pub/scm/com.bindlebinaries/iOSPortsExample.git
      ssh://ssh.scm.bindlebinaries.com/pub/scm/com.bindlebinaries/iOSPortsExample.git

   Downloading Source:

      $ git clone git://github.com/bindle/iOSPortsExample.git

   Git Branches:
      master - Current release of packages.
      next   - changes staged for next release
      pu     - proposed updates for next release
      xx/yy+ - branch for testing new changes before merging to 'pu' branch



ChangeLog

iOS Ports Example
Copyright (C) 2012 Bindle Binaries

0.3
        - Reverting supported architectures back to armv6, armv7. (syzdek)

0.2     2011-05-16
        - Updating project to Xcode4. (syzdek)
        - Fixing memory leak in example. (syzdek)

0.1	2010-11-30
        - Hello World (syzdek)


other/license.m

#import <iOSPorts/iOSPortsTypes.h>

const char iOSPorts_license[] =
{  // pkg_license
   0x0A, 0x69, 0x4F, 0x53, 0x20, 0x50, 0x6F, 0x72, 0x74, 0x73, 0x20, 0x45,
   0x78, 0x61, 0x6D, 0x70, 0x6C, 0x65, 0x0A, 0x0A, 0x43, 0x6F, 0x70, 0x79,
   0x72, 0x69, 0x67, 0x68, 0x74, 0x20, 0x28, 0x63, 0x29, 0x20, 0x32, 0x30,
   0x31, 0x30, 0x2C, 0x20, 0x42, 0x69, 0x6E, 0x64, 0x6C, 0x65, 0x20, 0x42,
   0x69, 0x6E, 0x61, 0x72, 0x69, 0x65, 0x73, 0x0A, 0x41, 0x6C, 0x6C, 0x20,
   0x72, 0x69, 0x67, 0x68, 0x74, 0x73, 0x20, 0x72, 0x65, 0x73, 0x65, 0x72,
   0x76, 0x65, 0x64, 0x2E, 0x0A, 0x0A, 0x40, 0x42, 0x49, 0x4E, 0x44, 0x4C,
   0x45, 0x5F, 0x42, 0x49, 0x4E, 0x41, 0x52, 0x49, 0x45, 0x53, 0x5F, 0x42,
   0x53, 0x44, 0x5F, 0x4C, 0x49, 0x43, 0x45, 0x4E, 0x53, 0x45, 0x5F, 0x53,
   0x54, 0x41, 0x52, 0x54, 0x40, 0x0A, 0x0A, 0x52, 0x65, 0x64, 0x69, 0x73,
   0x74, 0x72, 0x69, 0x62, 0x75, 0x74, 0x69, 0x6F, 0x6E, 0x20, 0x61, 0x6E,
   0x64, 0x20, 0x75, 0x73, 0x65, 0x20, 0x69, 0x6E, 0x20, 0x73, 0x6F, 0x75,
   0x72, 0x63, 0x65, 0x20, 0x61, 0x6E, 0x64, 0x20, 0x62, 0x69, 0x6E, 0x61,
   0x72, 0x79, 0x20, 0x66, 0x6F, 0x72, 0x6D, 0x73, 0x2C, 0x20, 0x77, 0x69,
   0x74, 0x68, 0x20, 0x6F, 0x72, 0x20, 0x77, 0x69, 0x74, 0x68, 0x6F, 0x75,
   0x74, 0x0A, 0x6D, 0x6F, 0x64, 0x69, 0x66, 0x69, 0x63, 0x61, 0x74, 0x69,
   0x6F, 0x6E, 0x2C, 0x20, 0x61, 0x72, 0x65, 0x20, 0x70, 0x65, 0x72, 0x6D,
   0x69, 0x74, 0x74, 0x65, 0x64, 0x20, 0x70, 0x72, 0x6F, 0x76, 0x69, 0x64,
   0x65, 0x64, 0x20, 0x74, 0x68, 0x61, 0x74, 0x20, 0x74, 0x68, 0x65, 0x20,
   0x66, 0x6F, 0x6C, 0x6C, 0x6F, 0x77, 0x69, 0x6E, 0x67, 0x20, 0x63, 0x6F,
   0x6E, 0x64, 0x69, 0x74, 0x69, 0x6F, 0x6E, 0x73, 0x20, 0x61, 0x72, 0x65,
   0x0A, 0x6D, 0x65, 0x74, 0x3A, 0x0A, 0x0A, 0x20, 0x20, 0x20, 0x2A, 0x20,
   0x52, 0x65, 0x64, 0x69, 0x73, 0x74, 0x72, 0x69, 0x62, 0x75, 0x74, 0x69,
   0x6F, 0x6E, 0x73, 0x20, 0x6F, 0x66, 0x20, 0x73, 0x6F, 0x75, 0x72, 0x63,
   0x65, 0x20, 0x63, 0x6F, 0x64, 0x65, 0x20, 0x6D, 0x75, 0x73, 0x74, 0x20,
   0x72, 0x65, 0x74, 0x61, 0x69, 0x6E, 0x20, 0x74, 0x68, 0x65, 0x20, 0x61,
   0x62, 0x6F, 0x76, 0x65, 0x20, 0x63, 0x6F, 0x70, 0x79, 0x72, 0x69, 0x67,
   0x68, 0x74, 0x0A, 0x20, 0x20, 0x20, 0x20, 0x20, 0x6E, 0x6F, 0x74, 0x69,
   0x63, 0x65, 0x2C, 0x20, 0x74, 0x68, 0x69, 0x73, 0x20, 0x6C, 0x69, 0x73,
   0x74, 0x20, 0x6F, 0x66, 0x20, 0x63, 0x6F, 0x6E, 0x64, 0x69, 0x74, 0x69,
   0x6F, 0x6E, 0x73, 0x20, 0x61, 0x6E, 0x64, 0x20, 0x74, 0x68, 0x65, 0x20,
   0x66, 0x6F, 0x6C, 0x6C, 0x6F, 0x77, 0x69, 0x6E, 0x67, 0x20, 0x64, 0x69,
   0x73, 0x63, 0x6C, 0x61, 0x69, 0x6D, 0x65, 0x72, 0x2E, 0x0A, 0x20, 0x20,
   0x20, 0x2A, 0x20, 0x52, 0x65, 0x64, 0x69, 0x73, 0x74, 0x72, 0x69, 0x62,
   0x75, 0x74, 0x69, 0x6F, 0x6E, 0x73, 0x20, 0x69, 0x6E, 0x20, 0x62, 0x69,
   0x6E, 0x61, 0x72, 0x79, 0x20, 0x66, 0x6F, 0x72, 0x6D, 0x20, 0x6D, 0x75,
   0x73, 0x74, 0x20, 0x72, 0x65, 0x70, 0x72, 0x6F, 0x64, 0x75, 0x63, 0x65,
   0x20, 0x74, 0x68, 0x65, 0x20, 0x61, 0x62, 0x6F, 0x76, 0x65, 0x20, 0x63,
   0x6F, 0x70, 0x79, 0x72, 0x69, 0x67, 0x68, 0x74, 0x0A, 0x20, 0x20, 0x20,
   0x20, 0x20, 0x6E, 0x6F, 0x74, 0x69, 0x63, 0x65, 0x2C, 0x20, 0x74, 0x68,
   0x69, 0x73, 0x20, 0x6C, 0x69, 0x73, 0x74, 0x20, 0x6F, 0x66, 0x20, 0x63,
   0x6F, 0x6E, 0x64, 0x69, 0x74, 0x69, 0x6F, 0x6E, 0x73, 0x20, 0x61, 0x6E,
   0x64, 0x20, 0x74, 0x68, 0x65, 0x20, 0x66, 0x6F, 0x6C, 0x6C, 0x6F, 0x77,
   0x69, 0x6E, 0x67, 0x20, 0x64, 0x69, 0x73, 0x63, 0x6C, 0x61, 0x69, 0x6D,
   0x65, 0x72, 0x20, 0x69, 0x6E, 0x20, 0x74, 0x68, 0x65, 0x0A, 0x20, 0x20,
   0x20, 0x20, 0x20, 0x64, 0x6F, 0x63, 0x75, 0x6D, 0x65, 0x6E, 0x74, 0x61,
   0x74, 0x69, 0x6F, 0x6E, 0x20, 0x61, 0x6E, 0x64, 0x2F, 0x6F, 0x72, 0x20,
   0x6F, 0x74, 0x68, 0x65, 0x72, 0x20, 0x6D, 0x61, 0x74, 0x65, 0x72, 0x69,
   0x61, 0x6C, 0x73, 0x20, 0x70, 0x72, 0x6F, 0x76, 0x69, 0x64, 0x65, 0x64,
   0x20, 0x77, 0x69, 0x74, 0x68, 0x20, 0x74, 0x68, 0x65, 0x20, 0x64, 0x69,
   0x73, 0x74, 0x72, 0x69, 0x62, 0x75, 0x74, 0x69, 0x6F, 0x6E, 0x2E, 0x0A,
   0x20, 0x20, 0x20, 0x2A, 0x20, 0x4E, 0x65, 0x69, 0x74, 0x68, 0x65, 0x72,
   0x20, 0x74, 0x68, 0x65, 0x20, 0x6E, 0x61, 0x6D, 0x65, 0x20, 0x6F, 0x66,
   0x20, 0x42, 0x69, 0x6E, 0x64, 0x6C, 0x65, 0x20, 0x42, 0x69, 0x6E, 0x61,
   0x72, 0x69, 0x65, 0x73, 0x20, 0x6E, 0x6F, 0x72, 0x20, 0x74, 0x68, 0x65,
   0x0A, 0x20, 0x20, 0x20, 0x20, 0x20, 0x6E, 0x61, 0x6D, 0x65, 0x73, 0x20,
   0x6F, 0x66, 0x20, 0x69, 0x74, 0x73, 0x20, 0x63, 0x6F, 0x6E, 0x74, 0x72,
   0x69, 0x62, 0x75, 0x74, 0x6F, 0x72, 0x73, 0x20, 0x6D, 0x61, 0x79, 0x20,
   0x62, 0x65, 0x20, 0x75, 0x73, 0x65, 0x64, 0x20, 0x74, 0x6F, 0x20, 0x65,
   0x6E, 0x64, 0x6F, 0x72, 0x73, 0x65, 0x20, 0x6F, 0x72, 0x20, 0x70, 0x72,
   0x6F, 0x6D, 0x6F, 0x74, 0x65, 0x20, 0x70, 0x72, 0x6F, 0x64, 0x75, 0x63,
   0x74, 0x73, 0x0A, 0x20, 0x20, 0x20, 0x20, 0x20, 0x64, 0x65, 0x72, 0x69,
   0x76, 0x65, 0x64, 0x20, 0x66, 0x72, 0x6F, 0x6D, 0x20, 0x74, 0x68, 0x69,
   0x73, 0x20, 0x73, 0x6F, 0x66, 0x74, 0x77, 0x61, 0x72, 0x65, 0x20, 0x77,
   0x69, 0x74, 0x68, 0x6F, 0x75, 0x74, 0x20, 0x73, 0x70, 0x65, 0x63, 0x69,
   0x66, 0x69, 0x63, 0x20, 0x70, 0x72, 0x69, 0x6F, 0x72, 0x20, 0x77, 0x72,
   0x69, 0x74, 0x74, 0x65, 0x6E, 0x20, 0x70, 0x65, 0x72, 0x6D, 0x69, 0x73,
   0x73, 0x69, 0x6F, 0x6E, 0x2E, 0x0A, 0x0A, 0x54, 0x48, 0x49, 0x53, 0x20,
   0x53, 0x4F, 0x46, 0x54, 0x57, 0x41, 0x52, 0x45, 0x20, 0x49, 0x53, 0x20,
   0x50, 0x52, 0x4F, 0x56, 0x49, 0x44, 0x45, 0x44, 0x20, 0x42, 0x59, 0x20,
   0x54, 0x48, 0x45, 0x20, 0x43, 0x4F, 0x50, 0x59, 0x52, 0x49, 0x47, 0x48,
   0x54, 0x20, 0x48, 0x4F, 0x4C, 0x44, 0x45, 0x52, 0x53, 0x20, 0x41, 0x4E,
   0x44, 0x20, 0x43, 0x4F, 0x4E, 0x54, 0x52, 0x49, 0x42, 0x55, 0x54, 0x4F,
   0x52, 0x53, 0x20, 0x22, 0x41, 0x53, 0x0A, 0x49, 0x53, 0x22, 0x20, 0x41,
   0x4E, 0x44, 0x20, 0x41, 0x4E, 0x59, 0x20, 0x45, 0x58, 0x50, 0x52, 0x45,
   0x53, 0x53, 0x20, 0x4F, 0x52, 0x20, 0x49, 0x4D, 0x50, 0x4C, 0x49, 0x45,
   0x44, 0x20, 0x57, 0x41, 0x52, 0x52, 0x41, 0x4E, 0x54, 0x49, 0x45, 0x53,
   0x2C, 0x20, 0x49, 0x4E, 0x43, 0x4C, 0x55, 0x44, 0x49, 0x4E, 0x47, 0x2C,
   0x20, 0x42, 0x55, 0x54, 0x20, 0x4E, 0x4F, 0x54, 0x20, 0x4C, 0x49, 0x4D,
   0x49, 0x54, 0x45, 0x44, 0x20, 0x54, 0x4F, 0x2C, 0x0A, 0x54, 0x48, 0x45,
   0x20, 0x49, 0x4D, 0x50, 0x4C, 0x49, 0x45, 0x44, 0x20, 0x57, 0x41, 0x52,
   0x52, 0x41, 0x4E, 0x54, 0x49, 0x45, 0x53, 0x20, 0x4F, 0x46, 0x20, 0x4D,
   0x45, 0x52, 0x43, 0x48, 0x41, 0x4E, 0x54, 0x41, 0x42, 0x49, 0x4C, 0x49,
   0x54, 0x59, 0x20, 0x41, 0x4E, 0x44, 0x20, 0x46, 0x49, 0x54, 0x4E, 0x45,
   0x53, 0x53, 0x20, 0x46, 0x4F, 0x52, 0x20, 0x41, 0x20, 0x50, 0x41, 0x52,
   0x54, 0x49, 0x43, 0x55, 0x4C, 0x41, 0x52, 0x0A, 0x50, 0x55, 0x52, 0x50,
   0x4F, 0x53, 0x45, 0x20, 0x41, 0x52, 0x45, 0x20, 0x44, 0x49, 0x53, 0x43,
   0x4C, 0x41, 0x49, 0x4D, 0x45, 0x44, 0x2E, 0x20, 0x49, 0x4E, 0x20, 0x4E,
   0x4F, 0x20, 0x45, 0x56, 0x45, 0x4E, 0x54, 0x20, 0x53, 0x48, 0x41, 0x4C,
   0x4C, 0x20, 0x42, 0x49, 0x4E, 0x44, 0x4C, 0x45, 0x20, 0x42, 0x49, 0x4E,
   0x41, 0x52, 0x49, 0x45, 0x53, 0x20, 0x42, 0x45, 0x20, 0x4C, 0x49, 0x41,
   0x42, 0x4C, 0x45, 0x20, 0x46, 0x4F, 0x52, 0x0A, 0x41, 0x4E, 0x59, 0x20,
   0x44, 0x49, 0x52, 0x45, 0x43, 0x54, 0x2C, 0x20, 0x49, 0x4E, 0x44, 0x49,
   0x52, 0x45, 0x43, 0x54, 0x2C, 0x20, 0x49, 0x4E, 0x43, 0x49, 0x44, 0x45,
   0x4E, 0x54, 0x41, 0x4C, 0x2C, 0x20, 0x53, 0x50, 0x45, 0x43, 0x49, 0x41,
   0x4C, 0x2C, 0x20, 0x45, 0x58, 0x45, 0x4D, 0x50, 0x4C, 0x41, 0x52, 0x59,
   0x2C, 0x20, 0x4F, 0x52, 0x20, 0x43, 0x4F, 0x4E, 0x53, 0x45, 0x51, 0x55,
   0x45, 0x4E, 0x54, 0x49, 0x41, 0x4C, 0x0A, 0x44, 0x41, 0x4D, 0x41, 0x47,
   0x45, 0x53, 0x20, 0x28, 0x49, 0x4E, 0x43, 0x4C, 0x55, 0x44, 0x49, 0x4E,
   0x47, 0x2C, 0x20, 0x42, 0x55, 0x54, 0x20, 0x4E, 0x4F, 0x54, 0x20, 0x4C,
   0x49, 0x4D, 0x49, 0x54, 0x45, 0x44, 0x20, 0x54, 0x4F, 0x2C, 0x20, 0x50,
   0x52, 0x4F, 0x43, 0x55, 0x52, 0x45, 0x4D, 0x45, 0x4E, 0x54, 0x20, 0x4F,
   0x46, 0x20, 0x53, 0x55, 0x42, 0x53, 0x54, 0x49, 0x54, 0x55, 0x54, 0x45,
   0x20, 0x47, 0x4F, 0x4F, 0x44, 0x53, 0x20, 0x4F, 0x52, 0x0A, 0x53, 0x45,
   0x52, 0x56, 0x49, 0x43, 0x45, 0x53, 0x3B, 0x20, 0x4C, 0x4F, 0x53, 0x53,
   0x20, 0x4F, 0x46, 0x20, 0x55, 0x53, 0x45, 0x2C, 0x20, 0x44, 0x41, 0x54,
   0x41, 0x2C, 0x20, 0x4F, 0x52, 0x20, 0x50, 0x52, 0x4F, 0x46, 0x49, 0x54,
   0x53, 0x3B, 0x20, 0x4F, 0x52, 0x20, 0x42, 0x55, 0x53, 0x49, 0x4E, 0x45,
   0x53, 0x53, 0x20, 0x49, 0x4E, 0x54, 0x45, 0x52, 0x52, 0x55, 0x50, 0x54,
   0x49, 0x4F, 0x4E, 0x29, 0x20, 0x48, 0x4F, 0x57, 0x45, 0x56, 0x45, 0x52,
   0x0A, 0x43, 0x41, 0x55, 0x53, 0x45, 0x44, 0x20, 0x41, 0x4E, 0x44, 0x20,
   0x4F, 0x4E, 0x20, 0x41, 0x4E, 0x59, 0x20, 0x54, 0x48, 0x45, 0x4F, 0x52,
   0x59, 0x20, 0x4F, 0x46, 0x20, 0x4C, 0x49, 0x41, 0x42, 0x49, 0x4C, 0x49,
   0x54, 0x59, 0x2C, 0x20, 0x57, 0x48, 0x45, 0x54, 0x48, 0x45, 0x52, 0x20,
   0x49, 0x4E, 0x20, 0x43, 0x4F, 0x4E, 0x54, 0x52, 0x41, 0x43, 0x54, 0x2C,
   0x20, 0x53, 0x54, 0x52, 0x49, 0x43, 0x54, 0x0A, 0x4C, 0x49, 0x41, 0x42,
   0x49, 0x4C, 0x49, 0x54, 0x59, 0x2C, 0x20, 0x4F, 0x52, 0x20, 0x54, 0x4F,
   0x52, 0x54, 0x20, 0x28, 0x49, 0x4E, 0x43, 0x4C, 0x55, 0x44, 0x49, 0x4E,
   0x47, 0x20, 0x4E, 0x45, 0x47, 0x4C, 0x49, 0x47, 0x45, 0x4E, 0x43, 0x45,
   0x20, 0x4F, 0x52, 0x20, 0x4F, 0x54, 0x48, 0x45, 0x52, 0x57, 0x49, 0x53,
   0x45, 0x29, 0x20, 0x41, 0x52, 0x49, 0x53, 0x49, 0x4E, 0x47, 0x20, 0x49,
   0x4E, 0x20, 0x41, 0x4E, 0x59, 0x20, 0x57, 0x41, 0x59, 0x0A, 0x4F, 0x55,
   0x54, 0x20, 0x4F, 0x46, 0x20, 0x54, 0x48, 0x45, 0x20, 0x55, 0x53, 0x45,
   0x20, 0x4F, 0x46, 0x20, 0x54, 0x48, 0x49, 0x53, 0x20, 0x53, 0x4F, 0x46,
   0x54, 0x57, 0x41, 0x52, 0x45, 0x2C, 0x20, 0x45, 0x56, 0x45, 0x4E, 0x20,
   0x49, 0x46, 0x20, 0x41, 0x44, 0x56, 0x49, 0x53, 0x45, 0x44, 0x20, 0x4F,
   0x46, 0x20, 0x54, 0x48, 0x45, 0x20, 0x50, 0x4F, 0x53, 0x53, 0x49, 0x42,
   0x49, 0x4C, 0x49, 0x54, 0x59, 0x20, 0x4F, 0x46, 0x0A, 0x53, 0x55, 0x43,
   0x48, 0x20, 0x44, 0x41, 0x4D, 0x41, 0x47, 0x45, 0x2E, 0x0A, 0x0A, 0x40,
   0x42, 0x49, 0x4E, 0x44, 0x4C, 0x45, 0x5F, 0x42, 0x49, 0x4E, 0x41, 0x52,
   0x49, 0x45, 0x53, 0x5F, 0x42, 0x53, 0x44, 0x5F, 0x4C, 0x49, 0x43, 0x45,
   0x4E, 0x53, 0x45, 0x5F, 0x45, 0x4E, 0x44, 0x40, 0x0A, 0x0A, 0x00
};




classes/iPad/AppDelegate_iPad.m
//
//  AppDelegate_iPad.m
//  iOSPorts
//
//  Created by David Syzdek on 11/22/10.
//  Copyright 2010 David M. Syzdek. All rights reserved.
//

#import "AppDelegate_iPad.h"
#import <iOSPorts/iOSPorts.h>

extern const char iOSPorts_license[];

@implementation AppDelegate_iPad

@synthesize window;
@synthesize portController;


#pragma mark -
#pragma mark Application lifecycle

- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions
{
   iOSPortsPackage        * portpkg;

   // Override point for customization after application launch.
   portpkg = [[iOSPortsPackage alloc] init];
   portpkg.name    = @"iOS Ports Example";
   portpkg.version = [[[NSBundle mainBundle] infoDictionary] valueForKey:@"CFBundleVersion"];
   portpkg.website = @"https://github.com/bindle/iOSPortsExample";
   portpkg.license = [NSString stringWithUTF8String:iOSPorts_license];

   self.portController = [[iOSPortsViewController alloc] init];
   [portController release];
   [portController addPackage:portpkg];
   [portController addPackageWithIdentifier:@"openssl"];
   [portController addPackageWithIdentifier:@"openldap"];
   [portController addPackageWithIdentifier:@"cyrus-sasl"];
   [portController addPackageWithIdentifier:@"pcre"];

   // Uncomment to set a custom frame for the PortsPackageView
   //portController.viewFrame = CGRectMake(0, 0, 320, 480);

   [window addSubview:portController.view];
   [window makeKeyAndVisible];

   [portpkg release];
   portpkg = nil;

   return YES;
}


- (void)applicationWillResignActive:(UIApplication *)application {
    /*
     Sent when the application is about to move from active to inactive state. This can occur for certain types of temporary interruptions (such as an incoming phone call or SMS message) or when the user quits the application and it begins the transition to the background state.
     Use this method to pause ongoing tasks, disable timers, and throttle down OpenGL ES frame rates. Games should use this method to pause the game.
     */
}


- (void)applicationDidBecomeActive:(UIApplication *)application {
    /*
     Restart any tasks that were paused (or not yet started) while the application was inactive.
     */
}


- (void)applicationWillTerminate:(UIApplication *)application {
    /*
     Called when the application is about to terminate.
     */
}


#pragma mark -
#pragma mark Memory management

- (void)applicationDidReceiveMemoryWarning:(UIApplication *)application {
    /*
     Free up as much memory as possible by purging cached data objects that can be recreated (or reloaded from disk) later.
     */
}


- (void)dealloc {
    [window release];
    [super dealloc];
}


@end


classes/iPhone/AppDelegate_iPhone.m
//
//  AppDelegate_iPhone.m
//  iOSPorts
//
//  Created by David Syzdek on 11/22/10.
//  Copyright 2010 David M. Syzdek. All rights reserved.
//

#import "AppDelegate_iPhone.h"
#import <iOSPorts/iOSPorts.h>

extern const char iOSPorts_license[];

@implementation AppDelegate_iPhone

@synthesize window;
@synthesize portController;


#pragma mark -
#pragma mark Application lifecycle

- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions
{
   iOSPortsPackage        * portpkg;

   // Override point for customization after application launch.
   portpkg = [[iOSPortsPackage alloc] init];
   portpkg.name    = @"iOS Ports Example";
   portpkg.version = [[[NSBundle mainBundle] infoDictionary] valueForKey:@"CFBundleVersion"];
   portpkg.website = @"https://github.com/bindle/iOSPortsExample";
   portpkg.license = [NSString stringWithUTF8String:iOSPorts_license];

   self.portController = [[iOSPortsViewController alloc] init];
   [portController release];
   [portController addPackage:portpkg];
   [portController addPackageWithIdentifier:@"openssl"];
   [portController addPackageWithIdentifier:@"openldap"];
   [portController addPackageWithIdentifier:@"cyrus-sasl"];
   [portController addPackageWithIdentifier:@"pcre"];

   // Uncomment to set a custom frame for the PortsPackageView
   //portController.viewFrame = CGRectMake(0, 0, 320, 480);

   [window addSubview:portController.view];
   [window makeKeyAndVisible];

   [portpkg release];
   portpkg = nil;

   return YES;
}


- (void)applicationWillResignActive:(UIApplication *)application {
    /*
     Sent when the application is about to move from active to inactive state. This can occur for certain types of temporary interruptions (such as an incoming phone call or SMS message) or when the user quits the application and it begins the transition to the background state.
     Use this method to pause ongoing tasks, disable timers, and throttle down OpenGL ES frame rates. Games should use this method to pause the game.
     */
}


- (void)applicationDidEnterBackground:(UIApplication *)application {
    /*
     Use this method to release shared resources, save user data, invalidate timers, and store enough application state information to restore your application to its current state in case it is terminated later. 
     If your application supports background execution, called instead of applicationWillTerminate: when the user quits.
     */
}


- (void)applicationWillEnterForeground:(UIApplication *)application {
    /*
     Called as part of  transition from the background to the inactive state: here you can undo many of the changes made on entering the background.
     */
}


- (void)applicationDidBecomeActive:(UIApplication *)application {
    /*
     Restart any tasks that were paused (or not yet started) while the application was inactive. If the application was previously in the background, optionally refresh the user interface.
     */
}


- (void)applicationWillTerminate:(UIApplication *)application {
    /*
     Called when the application is about to terminate.
     See also applicationDidEnterBackground:.
     */
}


#pragma mark -
#pragma mark Memory management

- (void)applicationDidReceiveMemoryWarning:(UIApplication *)application {
    /*
     Free up as much memory as possible by purging cached data objects that can be recreated (or reloaded from disk) later.
     */
}


- (void)dealloc {
    [window release];
    [super dealloc];
}


@end
